mongo-stream.js

```diff
@@ -0,0 +1,230 @@
+const BSON = require('bson');
+const bson = new BSON();
+const elasticsearch = require('elasticsearch');
+const fs = require('fs');
+const MongoClient = require('mongodb').MongoClient;
+
+
+class MongoStream {
+  constructor(db) {
+    this.esClient = new elasticsearch.Client({host: 'localhost:9200', apiVersion: '2.4'});
+    this.db = db;
+    this.changeStreams = {};
+  }
+
+  // constructs and returns a new MongoStream
+  static async init(url, options, dbName) {
```

Comment:
It would be more flexible to initialize the ES client here, using a supplied set of options. The constructor would then also take an ES client.